storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -204,6 +205,29 @@ private boolean hasReorgedFrom(
     return Optional.empty();
   }
 
+  /**
+   * Returns the fork info that applies based on the node's current slot, regardless of where the
+   * sync progress is up to.
+   *
+   * @return fork info based on the current time, not head block
+   */
+  public Optional<ForkInfo> getForkInfoAtCurrentTime() {
```

Comment:
Why do we return the next fork info when we're asked to get the current fork info?